src/main/java/duke/util/TaskList.java

```diff
@@ -0,0 +1,112 @@
+package duke.util;
+
+import duke.exception.DukeException;
+import duke.exception.ErrorMessage;
+import duke.task.Task;
+
+import java.util.ArrayList;
+
+/**
+ * Contains a list of tasks and has operations to manipulate tasks in the list.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Creates an empty task list.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Creates a task list with the specified data.
+     *
+     * @param tasks List of tasks to be added to the task list.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = new ArrayList<>(tasks);
+    }
+
+    public int getSize() {
+        return tasks.size();
+    }
+
+    /**
+     * Returns the task at the specified index.
+     *
+     * @param index One-based index of the task to be retrieved.
+     * @return Task at index - 1.
+     * @throws DukeException if the index is invalid.
+     */
+    public Task getTask(int index) throws DukeException {
+        int zeroBasedIndex = index - 1;
+        if (zeroBasedIndex < 0 || zeroBasedIndex >= tasks.size()) {
+            throw new DukeException(ErrorMessage.MESSAGE_INVALID_INDEX);
+        }
```

Comment:
Good job in considering these errors and explicitly naming the variable `zeroBasedIndex` to avoid confusion!